### PR TITLE
Upgrade Guava dependency due CVE 2020-8908 and other library upgrades

### PR DESCRIPTION
**Pull request by GittiMcHub**
Hi @erosb,

please be patient with me as this is my first contribution to an OpenSource project. I checked the [contributing guidelines](https://github.com/everit-org/json-schema/blob/master/CONTRIBUTING.md), but I raise a PR not actually for a bug report - I just would like to upgrade the dependencies. 

The OWASP Dependency check reported issues that I tracked down being being the [Guava Library](https://mvnrepository.com/artifact/com.google.guava/guava) in Version [29.0-jre](https://mvnrepository.com/artifact/com.google.guava/guava/29.0-jre) being vulnerble to [CVE-2020-8908](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-8908), which is referenced in [json-schema/core/pom.xml](https://github.com/GittiMcHub/json-schema/blob/13716721902f5db86c0d2603c26a21fe927c4ded/core/pom.xml#L227)
```
        <dependency>
            <groupId>com.google.guava</groupId>
            <artifactId>guava</artifactId>
            <version>29.0-jre</version>
            <scope>test</scope>
        </dependency>
```

I then saw additional library upgrades available and bumped the version to the latest - due breaking changes in the Mockito library, I had to do a minor fix in the tests as well.

Using 
```
mvn clean install
```
The changes seem to work. 

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for org.everit.json.schema.parent 0.0.0-develop:
[INFO] 
[INFO] everit-org/json-schema ............................. SUCCESS [ 13.401 s]
[INFO] org.everit.json.schema.parent ...................... SUCCESS [  0.037 s]
[INFO] org.everit.json.schema.tests.parent ................ SUCCESS [  0.004 s]
[INFO] org.everit.json.schema.tests.vanilla ............... SUCCESS [  3.897 s]
[INFO] org.everit.json.schema.tests.android ............... SUCCESS [  3.465 s]
[INFO] org.everit.json.schema.tests.slim .................. SUCCESS [  3.445 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  24.943 s
[INFO] Finished at: 2022-01-20T08:46:02-08:00
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

File: core/src/test/java/org/everit/json/schema/loader/JsonValueTest.java [hunk 1]
```diff
@@ -1,27 +1,5 @@
 package org.everit.json.schema.loader;
 
-import static java.util.Arrays.asList;
```
Comment (erosb) [inline core/src/test/java/org/everit/json/schema/loader/JsonValueTest.java:3]: Is this import reordering necessary? If there is no significant change then can you please revert it?

**Comment by erosb** [comment]
Hello @GittiMcHub thank you for raising the PR, and also big kudos to raise your first open-source PR !! :+1:  :tada: 
I also appreciate that you read the contribution guidelines. Most people don't do that.

I left one comment on the changes. Beside that, since tests pass, this is good to merge.

File: core/src/test/java/org/everit/json/schema/loader/JsonValueTest.java (hunk 1 above)
Comment (GittiMcHub) [reply core/src/test/java/org/everit/json/schema/loader/JsonValueTest.java:3]: I have to admit that I just let the IDE autofixing the unused imports of 
import java.util.ArrayList;
import java.util.Arrays;

I can revert the changes and just remove the unused imports

**Comment by GittiMcHub** [comment]
Hi @erosb,

first of all thanks for the very quick reply! I reverted the reordering and just commited the actual changes - please check if it's now as expected

**Review by erosb — APPROVED** [review]

**Comment by erosb** [comment]
@GittiMcHub great, thanks, merged. In case you are interested in contributing to other open-source java projects as well, you may look into [Hazelcast](https://github.com/hazelcast/hazelcast). If it looks like a project that you would be happy to contribute to, then I can help you with getting started with it.

**Comment by GittiMcHub** [comment]
Hi @erosb,
wow that was quick, thank you very much! 
The project looks very interesting indeed, will definitely check it out. But to be honest, I'm afraid that I don't have time for contributing in projects I'm not regulary using (at least for the time being now). I'll let you know when it comes across.
We use Apache Beam/DataFlow at work, where I might can contribute in future too.

**Pull request merged**